### PR TITLE
Features/employee details page

### PR DESCRIPTION
**Pull request by AlexeyGrishechkin**

### CODE REVIEW

File: src/pages/HomepageManager/index.tsx [hunk 1]
```diff
@@ -112,7 +108,7 @@ const HomepageManager = ({
       }
     } else {
       history.push(
-        `${createSearchString({ ...getUrlParams(), [`${selectName}`]: value })}`
+        `${createSearchString({  [`${selectName}`]: value })}`
```
Comment (alexanderchernetsky) [inline src/pages/HomepageManager/index.tsx:111]: Is this change related to the new page? If NOT - it shouldn't be a part of this PR!

Comment (alexanderchernetsky) [reply src/pages/HomepageManager/index.tsx:111]: It applies to other changes below

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,16393 @@
+{
```
Comment (alexanderchernetsky) [inline package-lock.json:1]: this file should be excluded from the PR

File: src/pages/HomepageManager/index.tsx [hunk 3]
```diff
@@ -77,11 +77,7 @@ const columns = [
           <StyledActionColButton
             type="primary"
             htmlType="button"
-            href={`/view-profile/${data.id}`}
-            onClick={(event) => {
-              event.preventDefault();
-              alert("Coming soon...");
-            }}
+            href={`/employee-details/${data.id}`}
```
Comment (alexanderchernetsky) [inline src/pages/HomepageManager/index.tsx:80]: The functionality is NOT working at all. You MUST check how the app works before creating the PR! https://streamable.com/ga3zxx

**Review by alexanderchernetsky — CHANGES_REQUESTED** [review]
I see your idea of re-using an existing page. But in this case I wanted you to create a new one. That's for the sake of exercise.  So please create new page, try to think up new design etc.

File: src/pages/HomepageManager/index.tsx (hunk 1 above)
Comment (AlexeyGrishechkin) [reply src/pages/HomepageManager/index.tsx]: rollbacked this

File: package-lock.json (hunk 2 above)
Comment (AlexeyGrishechkin) [reply package-lock.json]: exclude this file

File: src/pages/HomepageManager/index.tsx (hunk 3 above)
Comment (AlexeyGrishechkin) [reply src/pages/HomepageManager/index.tsx:80]: fixed

File: src/components/EmployeeInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-start;
+  align-items: flex-start;
+  width: 100%;
+  box-sizing: border-box;
+  padding: 70px 50px;
+  border-right: 2px solid ${ax("border-color")};
+  position: relative;
+  background: url("https://st4.cannypic.com/thumbs/42/422973_632_canny_pic.jpg")
```
Comment (alexanderchernetsky) [inline src/components/EmployeeInfo.tsx:20]: You should upload the image locally

File: src/components/EmployeeInfo.tsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+  width: 200px;
+`;
+
+const StyledActionColButton = styled(StyledButton)`
+  margin: 0 auto;
+  width: 100%;
+`;
+
+interface IEmployeeInfo {
+  imageSrc?: string;
```
Comment (alexanderchernetsky) [inline src/components/EmployeeInfo.tsx:108]: Why have you marked all the fields as optional?

File: src/components/EmployeeInfo.tsx [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+interface IEmployee {
+  link: string;
+  name: string;
+  profile_name: string;
+}
+
+const EmployeeInfo = ({
+  imageSrc = noLogoImage,
+  fullName = "Full Name",
```
Comment (alexanderchernetsky) [inline src/components/EmployeeInfo.tsx:130]: Most of these default values just don't have sense, thus, should be removed

File: src/components/EmployeeInfo.tsx [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+  rating = 1,
+  birth = "24 may 2020",
+  age = 25,
+  yearsOfExperience = 0,
+  email = "example@mail.ru",
+  shortDescription = "Employee short description.",
+  links = [],
+}: IEmployeeInfo): React.ReactElement => {
+  return (
+    <Wrapper>
```
Comment (alexanderchernetsky) [inline src/components/EmployeeInfo.tsx:143]: After looking into the markup, I've noticed that almost all tags are 'div'. Do you know any other tags? Have you heard about 'semantic markup'? Please learn about it and rewrite the markup in more semantic way.

File: src/components/EmployeeInfo.tsx [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+          >
+            Contact {name}
+          </StyledActionColButton>
+        </ActionColWrapper>
+        {links.map((item: IEmployee, index: number) => {
+          return (
+            <ActionColWrapper>
+              <div style={{ margin: "0 auto" }}>
+                <SocMediaLink
+                  key={index}
```
Comment (alexanderchernetsky) [inline src/components/EmployeeInfo.tsx:180]: the key is applied to the wrong element. You will notice the warning in the console

File: src/components/EmployeeInfo.tsx [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+            htmlType="button"
+            href={`mailto:${email}`}
+          >
+            Contact {name}
+          </StyledActionColButton>
+        </ActionColWrapper>
+        {links.map((item: IEmployee, index: number) => {
+          return (
+            <ActionColWrapper>
+              <div style={{ margin: "0 auto" }}>
```
Comment (alexanderchernetsky) [inline src/components/EmployeeInfo.tsx:178]: Don't ever use inline styles! Please use styled components for styling

File: src/pages/HomepageManager/index.tsx [hunk 10]
```diff
@@ -123,6 +119,7 @@ const HomepageManager = ({
       delete urlParams.searchPhrase;
       history.push(`${createSearchString(urlParams)}`);
     } else {
+      delete urlParams.page;
```
Comment (alexanderchernetsky) [inline src/pages/HomepageManager/index.tsx:122]: this bugfix should not be a part of this PR. Please remove

File: src/pages/EmployeeDetails/index.tsx [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+import {
+  EmployeeHomepageContent,
+  SpinnerWrapper,
+} from "../../styled-components/HomepageEmployee";
+import EmployeeInfo from "../../components/EmployeeInfo";
+
+const EmployeeDetails = ({
+  location,
+}: RouteComponentProps): React.ReactElement => {
+  const id = location.pathname.replace("/employee-details/", "");
```
Comment (alexanderchernetsky) [inline src/pages/EmployeeDetails/index.tsx:18]: This replace is unnecessary. You can get id from params. Please rewrite

File: src/pages/EmployeeDetails/index.tsx [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+  const user = feedBackStore.employeeData;
+
+  useEffect(() => {
+    feedBackStore.getSpecificEmployeeData(id);
+  }, [id]);
+
+  return (
+    <Layout>
+      {!loginStore?.user ? (
```
Comment (alexanderchernetsky) [inline src/pages/EmployeeDetails/index.tsx:27]: please simplify the logic. Remove ! operator and then rewrite ternary op.

File: src/pages/EmployeeDetails/index.tsx [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+        <SpinnerWrapper>
+          <Spin size="large" />
+        </SpinnerWrapper>
+      ) : (
+        <>
+          <Header pageTitle="Employee Details" />
```
Comment (alexanderchernetsky) [inline src/pages/EmployeeDetails/index.tsx:33]: I think this header can be shown anyways

File: src/components/EmployeeInfo.tsx [hunk 14]
```diff
@@ -0,0 +1,195 @@
...
+  return (
+    <Wrapper>
+      <Photo>
+        <img src={imageSrc} alt="user" />
+      </Photo>
+      <MainInfoWrapper>
+        <ProfileInfo>
+          <FullName>{fullName}</FullName>
+          <JobInfo>
+            {capitalize(focus)} <JobName>— {jobTitle}</JobName>
```
Comment (alexanderchernetsky) [inline src/components/EmployeeInfo.tsx:151]: Could you please remove the hyphen? It should look better without it.

**Review by alexanderchernetsky — CHANGES_REQUESTED** [review]

**Comment by alexanderchernetsky** [comment]
You have a flickering issue on the page. https://streamable.com/u9ongs You can use network throttling to see it clearly. Please fix the issue.

File: src/components/EmployeeInfo.tsx (hunk 4 above)
Comment (AlexeyGrishechkin) [reply src/components/EmployeeInfo.tsx]: done

File: src/components/EmployeeInfo.tsx (hunk 5 above)
Comment (AlexeyGrishechkin) [reply src/components/EmployeeInfo.tsx:112]: fixed

File: src/components/EmployeeInfo.tsx (hunk 6 above)
Comment (AlexeyGrishechkin) [reply src/components/EmployeeInfo.tsx]: fixed

File: src/components/EmployeeInfo.tsx (hunk 7 above)
Comment (AlexeyGrishechkin) [reply src/components/EmployeeInfo.tsx:146]: fixed

File: src/components/EmployeeInfo.tsx (hunk 8 above)
Comment (AlexeyGrishechkin) [reply src/components/EmployeeInfo.tsx]: fixed

File: src/pages/HomepageManager/index.tsx (hunk 10 above)
Comment (AlexeyGrishechkin) [reply src/pages/HomepageManager/index.tsx]: removed

File: src/pages/EmployeeDetails/index.tsx (hunk 11 above)
Comment (AlexeyGrishechkin) [reply src/pages/EmployeeDetails/index.tsx]: fixed

File: src/pages/EmployeeDetails/index.tsx (hunk 12 above)
Comment (AlexeyGrishechkin) [reply src/pages/EmployeeDetails/index.tsx]: fixed

File: src/pages/EmployeeDetails/index.tsx (hunk 13 above)
Comment (AlexeyGrishechkin) [reply src/pages/EmployeeDetails/index.tsx]: fixed

File: src/components/EmployeeInfo.tsx (hunk 14 above)
Comment (AlexeyGrishechkin) [reply src/components/EmployeeInfo.tsx]: fixed

File: src/components/EmployeeInfo.tsx (hunk 4 above)
Comment (alexanderchernetsky) [reply src/components/EmployeeInfo.tsx]: the image doesn't exist locally. And it's the reason why the app crashes.

File: src/components/EmployeeInfo.tsx [hunk 15]
```diff
@@ -0,0 +1,197 @@
...
+import styled from "styled-components";
+import moment from "moment";
+import ax from "../styled-components/accessor";
+import noLogoImage from "../images/no_logo_image.png";
+import ProfileBackground from "../images/profile_background.jpg";
+import SocMediaLink from "./SocMediaLink";
+import capitalize from "../helpers/capitalize";
+import { StyledButton } from "../styled-components/FeedbackPage";
+
+const Wrapper = styled.main`
```
Comment (alexanderchernetsky) [inline src/components/EmployeeInfo.tsx:11]: Remove empty space from the top of the page https://ibb.co/Y0ZL0VY

File: src/components/EmployeeInfo.tsx [hunk 16]
```diff
@@ -0,0 +1,197 @@
...
+
+const TextWrapper = styled.article`
+  font-size: 16px;
+  line-height: 22px;
+  color: ${ax("user-info-text-color")};
+  width: 70%;
+  margin-top: 30px;
+`;
+
+const LinksWrapper = styled.section`
```
Comment (alexanderchernetsky) [inline src/components/EmployeeInfo.tsx:88]: you're rendering a list. Isn't it semantically better to use ul and li tags?

File: src/pages/EmployeeDetails/index.tsx (hunk 12 above)
Comment (alexanderchernetsky) [reply src/pages/EmployeeDetails/index.tsx]: You made a mistake after re-writing. Now you have 100px of redundant space at the top of your page. It belongs to SpinnerWrapper. You shouldn't have removed ternary operator. I asked you to re-write it.

**Review by alexanderchernetsky — CHANGES_REQUESTED** [review]
You still have flickering on your page (image placeholder is shown initially) Please fix it.

File: src/components/EmployeeInfo.tsx (hunk 4 above)
Comment (AlexeyGrishechkin) [reply src/components/EmployeeInfo.tsx]: fixed

File: src/pages/EmployeeDetails/index.tsx (hunk 12 above)
Comment (AlexeyGrishechkin) [reply src/pages/EmployeeDetails/index.tsx]: fixed

File: src/components/EmployeeInfo.tsx [hunk 17]
```diff
@@ -0,0 +1,197 @@
...
+  margin-top: 20px;
+`;
+
+const ProfileInfo = styled.article`
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-start;
+`;
+
+const JobInfo = styled.h3`
```
[the remark this answers is not in the document]
Comment (AlexeyGrishechkin) [reply src/components/EmployeeInfo.tsx]: fixed

File: src/components/EmployeeInfo.tsx (hunk 15 above)
Comment (AlexeyGrishechkin) [reply src/components/EmployeeInfo.tsx:10]: done

File: src/components/EmployeeInfo.tsx (hunk 16 above)
Comment (AlexeyGrishechkin) [reply src/components/EmployeeInfo.tsx]: fixed

**Review by alexanderchernetsky — APPROVED** [review]

**Pull request merged**